### DISCUSSION TITLE
Mocking with MSW stops working when replacing document cache with Graphcache

### DISCUSSION THREAD

**Discussion by zorzysty** [urql-graphql/urql#3586] [Q&A] [answered]
My setup uses MSW for mocking requests. It works well with document cache. But as soon as I switch to graphcache, data is no longer provided to components by `useQuery` hook. It goes straight from `undefined` when loading, to `null` when request completes.

The only difference in code is changing where I import `cacheExchange` from and changing `cacheExchange` to `cacheExchange({})` in the `exchanges` array.

What could be the reason for that?

**Comment by JoviDeCroock** [comment, marked as answer]
The reason is most likely that your data is missing the `__typename` property, i.e. the selection-set is not fully completed. You can test this out by leveraging the `logger` property on the construction of your cache-exchange.

**Reply by zorzysty** [reply]
Thank you! That helped me solve the problem.
